site/docs/cow-and-mor.md

```diff
@@ -0,0 +1,195 @@
+                         delete.3_2 (update)
+```
+
+If P2 commits after P1:
+
+- if `delete.3_2` is a position delete, the update has to be retried from beginning because the old file is removed 
+- if `delete.3_2` is an equality delete, the update can succeed after retry because `delete.3_2` will also affect `data.5`
+
+If P1 commits after P2:
+
+- if `delete.3_2` is a position delete, the compaction has to be retried from beginning because it removes `data.3` which undeletes `delete.3_2`. 
+- if `delete.3_2` is an equality delete, the compaction has to be retried from beginning because it writes `data.5` with a higher sequence number than `delete.3_2` which undeletes `delete.3_2`
+
+To solve the issue, we allow MERGE to preserve sequence number. 
+With this additional feature, we do not need to retry compaction if P1 commits after P2 and `delete.3_2` is an equality delete,
+because `data.5` has a lower sequence number than `delete.3_2` so the deletion is still effective.
+
+This is particularly useful in the CDC streaming case, which is discussed more in the next section.
+
+### MoR Use Case Analysis
+
+Now we analyze 2 common MoR use cases in Iceberg.
+
+#### Use case 1: GDPR updates
+
+From time to time, users would like to request the removal of all their historical data which has the following characteristics:
+
+1. there is a relatively long grace period allowed before hard data deletion, like 1-3 months
+2. because of that, requests could potentially be processed in batch
+2. the data to delete spans across multiple partitions in a table
+
+CoW is a viable option for users with GDPR requirements, but MoR can also be used for such use case if there are specific SLA requirements around aspects such as logical delete latency.
+At write time, the system can can make the following choice:
+
+1. write global equality deletes on `userId`.
+2. if possible, figure out the related partition predicates and write partition equality deletes.
+3. perform a scan and write position deletes, the time taken would be similar to a query using the same predicate.
+
+Note that faster write would usually mean slower read and more compaction needed later.
+Here are the possible compaction actions:
+
+1. run CONVERT to remove equality deletes and avoid out-of-memory risk
+2. run REWRITE if the system would like to batch more deletes without doing a MERGE
+3. run MERGE or data compaction to clean up deletes
+
+The choice of the delete type to write and compactions to perform depends on the actual SLA of the system.
+Iceberg offers all these tools for users to flexibly customize based on their own needs.
+
+#### Use case 2: CDC Streaming
+
+In CDC streaming, for each new event, writer writes a new data row in memory (or temp storage like RocksDB) with:
+
+- a new equality delete row to remove any duplicated rows from existing Iceberg data files in storage, the equality constraint is formulated based on the primary key (Iceberg identifier fields)
```

Comment:
Maybe hard stop after storage and start "The equality constraint is..." as a new sentence?